tools/trace_bind.py

```diff
@@ -0,0 +1,430 @@
+#!/usr/bin/env bcc-py
+#
+# tcpbind       Trace TCP binds()s.
+#               For Linux, uses BCC, eBPF. Embedded C.
+#
+# based on tcpconnect utility from Brendan Gregg's suite.
+#
+# USAGE: trace_bind [-h] [-c] [-t] [-E] [-p PID] [-P PORT [PORT ...]]
+#
+# it is reporting socket options set before the bins call
+# impacting system call behavior:
+# SOL_IP     IP_FREEBIND              F....
+# SOL_IP     IP_TRANSPARENT           .T...
+# SOL_IP     IP_BIND_ADDRESS_NO_PORT  ..N..
+# SOL_SOCKET SO_REUSEADDR             ...R.
+# SOL_SOCKET SO_REUSEPORT             ....r
+#
+# SO_BINDTODEVICE interface is reported as "BOUND_IF" index
+#
+# This uses dynamic tracing of kernel functions, and will need to be updated
+# to match kernel changes.
+#
+
+from __future__ import print_function
+from bcc import BPF, DEBUG_SOURCE
+from bcc.utils import printb
+import argparse
+from socket import inet_ntop, AF_INET, AF_INET6
+from struct import pack
+from time import sleep
+
+# arguments
+examples = """examples:
+    ./trace_bind           # trace all TCP bind()s
+    ./trace_bind -t        # include timestamps
+    ./trace_bind -p 181    # only trace PID 181
+    ./trace_bind -P 80     # only trace port 80
+    ./trace_bind -P 80,81  # only trace port 80 and 81
+    ./trace_bind -U        # include UID
+    ./trace_bind -u 1000   # only trace UID 1000
+    ./trace_bind -E        # report bind errors
+    ./trace_bind -c        # count bind per src ip
```

Comment:
I try to reserve "-c" for cgroup path, could you use something different? Maybe use a long form like `--per-src-ip`?

Reply:
removing "-c", leaving "--count" option only